### PR TITLE
Generic version of Command

### PR DESCRIPTION
**Pull request by wongcain**

### CODE REVIEW

File: grox-sample-rx/src/main/java/com/groupon/grox/sample/RefreshColorCommand.java [hunk 1]
```diff
@@ -17,44 +17,40 @@
...
-  private Observable<? extends Action> refreshColor(){
-      return getColorFromServer()
-          .subscribeOn(io())
-          .map(ChangeColorAction::new)
-          .cast(Action.class)
-          .onErrorReturn(ErrorAction::new);
+  private Observable<Action<State>> refreshColor() {
+    return getColorFromServer()
+        .subscribeOn(io())
+        .map(color -> (Action<State>) new ChangeColorAction((color)))
```
Comment (alin-turcu) [inline grox-sample-rx/src/main/java/com/groupon/grox/sample/RefreshColorCommand.java]: @wongcain in my opinion the previous version of the chain looks better. It also allows us to use lambda expressions. @stephanenicolas  what do you think?

**Comment by wongcain** [comment]
How about this?

**Comment by alin-turcu** [comment]
@wongcain. The changes look good. We will test this and come back with an answer

**Comment by chenthorne** [comment]
This is working for us, we created similar classes locally and haven't had any issues.

**Pull request merged**